dnf5/commands/download/download.cpp

```diff
@@ -146,20 +155,34 @@ void DownloadCommand::run() {
         }
     }
 
-    if (!download_pkgs.empty()) {
-        libdnf5::repo::PackageDownloader downloader(ctx.base);
-
-        // for download command, we don't want to mark the packages for removal
-        downloader.force_keep_packages(true);
+    if (download_pkgs.empty()) {
+        return;
+    }
 
-        for (auto & [nevra, pkg] : download_pkgs) {
-            downloader.add(pkg);
+    if (url_option->get_value()) {
+        for (auto & [nerva, pkg] : download_pkgs) {
+            auto urls = pkg.get_remote_locations();
+            if (urls.empty()) {
+                std::cout << "Failed to get mirror for package: " << pkg.get_name() << std::endl;
+                continue;
+            }
```

Comment:
I don't think this should go to `stdout`. Either just log it like old dnf:
```cpp
ctx.base.get_logger()->warning("Failed to get mirror for package: \"{}\"", pkg.get_name());
```

or maybe even better we could try making it strict and throw an exception. 
```cpp
libdnf_assert(urls.empty(), "Failed to get mirror for package: \"{}\"", pkg.get_name());
```
I think this basically shouldn't happen but if it turns out to be a problem we can change it later.

Reply:
Used the libdnf_assert option as suggested